econet: update en75_chboot to use OpenWrt board_name

cjdelisle opened the pull request:
Instead of using the name from /proc/cpuinfo, use board_name from /lib/functions.sh
Also fix a goof up in the name of Zyxel PMG5617GA.

---
This addresses issue noted in thread: https://github.com/openwrt/openwrt/pull/20580#discussion_r2541270060
Thanks @Rondom for the advice.

cjdelisle committed:
econet: update en75_chboot to use OpenWrt board_name

Instead of using the name from /proc/cpuinfo, use board_name from
/lib/functions.sh

Signed-off-by: Caleb James DeLisle <cjd@cjdns.fr>
Link: https://github.com/openwrt/openwrt/pull/21023
Signed-off-by: Robert Marko <robimarko@gmail.com>

Rondom commented on target/linux/econet/dts/en751221_zyxel_pmg5617ga.dts:
```diff
@@ -5,7 +5,7 @@
 / {
 	model = "Zyxel PMG5617GA";
-	compatible = "tplink,zyxel-pmg5617ga", "econet,en751221";
+	compatible = "zyxel,pmg5617ga", "econet,en751221";
```
This will affect sysupgrade

981213 commented on target/linux/econet/base-files/sbin/en75_chboot:
```diff
@@ -99,32 +101,32 @@ switch() {
-    machine=$(sed -n -e 's/^machine\s\+:\s\+//p' < /proc/cpuinfo)
-    if [ "$machine" = "TP-Link Archer VR1200v (v2)" ]; then
+    machine=$(board_name)
+    if [ "$machine" = "tplink,archer-vr1200v-v2" ]; then
```
Hi! It would be better if you also replace this if-elif-else sequence with a switch/case statement, like how other scripts (e.g. [this one](https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/generic/base-files/etc/board.d/02_network) ) handles board-level differences.

cjdelisle committed:
econet: fix wrong board name for Zyxel PMG5617GA

Board name was specified incorrectly in the DT, fix.
This bug was introduced in #20580 (31f5fc8fea8931f5) which was
merged last week, and econet is still a mostly experimental target,
so it is considered unnecessary to update SUPPORTED_DEVICES.

Signed-off-by: Caleb James DeLisle <cjd@cjdns.fr>
Link: https://github.com/openwrt/openwrt/pull/21023
Signed-off-by: Robert Marko <robimarko@gmail.com>
